src/twisted/internet/unix.py

```diff
@@ -171,10 +171,22 @@ def doRead(self):
             else:
                 return main.CONNECTION_LOST
 
-        if ancillary:
-            ancillaryData = ancillary[0][2]
-            fdCount = len(ancillaryData) // 4
-            fds = struct.unpack('i'*fdCount, ancillaryData)
+        for cmsgLevel, cmsgType, cmsgData in ancillary:
+            if (cmsgLevel != socket.SOL_SOCKET or
+                cmsgType != sendmsg.SCM_RIGHTS):
+                log.msg(
+                    format=(
+                        "%(protocolName)s (on %(hostAddress)r) "
+                        "received unsupported ancillary data "
+                        "(level=%(cmsgLevel)r, type=%(cmsgType)r) "
+                        "from %(peerAddress)r."),
+                    hostAddress=self.getHost(), peerAddress=self.getPeer(),
+                    protocolName=self._getLogPrefix(self.protocol),
+                    cmsgLevel=cmsgLevel, cmsgType=cmsgType,
+                    )
+                continue
+            fdCount = len(cmsgData) // 4
+            fds = struct.unpack('i'*fdCount, cmsgData)
```

Comment:
(optional)

Given the structure of this code, I'm excited to introduce support for receiving other kinds of ancillary data.  It looks like it should be quite straightforward to do so.  If you wanted, you could take one additional step towards this and improve the way dispatch is done here.  For example, the fd handling logic could move into an `_ancillary_SOL_SOCKET_SCM_RIGHTS` method which is invoked via dict-style lookup.  The fall-through case would then be the log-and-drop logic.

However, for the single case that's actually supported now, the code is perfectly fine as-is.  Also, I know there's also `SCM_CREDENTIALS` (and googling turns up some really obscure stuff like `SCM_WIFI_STATUS`).  I don't know if there's anything else useful beyond these two ... If there are only two then dict-based dispatch is not _much_ of a win (certainly not performance; perhaps still readability).